FullstackAcademy/Study-Saturday-Fullstack
# Overhaul: Cycle 3

### **Upstream PRs to approve _first_:**
1. https://github.com/FullstackAcademy/Study-Saturday-Fullstack/pull/4
2. https://github.com/FullstackAcademy/Study-Saturday-Fullstack/pull/5
3. https://github.com/FullstackAcademy/Study-Saturday-Fullstack/pull/6

### **Branch includes the following:**
- [x] Updates README to highlight cycle step and outline instructions
- [x] Adds `POST /api/students` route solution code from Cycle 0
  - Tested in [previous PR](https://github.com/FullstackAcademy/Study-Saturday-Fullstack/pull/5)
- [x] Renders `NewStudentForm` in `Main` and adds solution code from Cycle 1
  - Tested in [previous PR](https://github.com/FullstackAcademy/Study-Saturday-Fullstack/pull/6)
- [x] Adds solution code from Cycle 2
  - `NewStudentForm` component: local `state`, `handleChange` and `handleSubmit`

### **Steps to test:**
- Clone repo (`npm i`, `npm run seed`, `npm run start`)
- `git checkout dependency-updates-cycle-3`
- Visit http://localhost:3000/ (open console)
- Toggle form by clicking the "Add New Student" button
- Fill in the form and submit
- Refresh the page to see the submitted student in the list of students
  - _Note: The next Cycle will push the newly created `student` into `state`._

### **Important!**
The next Cycle, `cycle-4-bonus`, did ***not*** have a starting point so it's completely new.
Since a PR cannot be made, kindly test that the steps from Cycle 3 have been incorporated:
- `git checkout cycle-4-bonus`
  - **Test the following:**
    - After submitting a new student via the form, see the student immediately added to the list
    - Click on that student to see their details
